src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -79,7 +79,11 @@ void Runnable::run_next_measurement(size_t thread_id, Callback<Error> cb,
         report::Entry entry;
         entry["test_keys"] = *test_keys;
         entry["test_keys"]["client_resolver"] = resolver_ip;
-        entry["input"] = next_input;
+        if (!(*test_keys)["input"].is_null()) {
+            entry["input"] = (*test_keys)["input"];
+        } else {
+            entry["input"] = next_input;
+        }
```

Comment:
As an alternative solution, what about this: there is a method in the base class that takes in input the `next_input` and returns in output a `report::Entry`. Such method has as default implementation:

```C++
report::Entry BaseClass::method_name(const std::string &next_input) {
    return next_input; /* Works because a string can be used to construct a valid JSON */
}
```

Then, only for this test, we override said method to construct the input representation we want.

Finally, in the code here we do:

```C++
    entry["input"] = method_name(next_input);
```

?